https://reviews.llvm.org/D143017

[Libomptarget] Fix disabling amdgpu on non-Linux.

Previously, on non-Linux, amdgpu would get enabled whatever the CPU
architecture.

Index: llvm-toolchain-16_16.0.0~+rc4/openmp/libomptarget/plugins/amdgpu/CMakeLists.txt
===================================================================
--- llvm-toolchain-16_16.0.0~+rc4.orig/openmp/libomptarget/plugins/amdgpu/CMakeLists.txt
+++ llvm-toolchain-16_16.0.0~+rc4/openmp/libomptarget/plugins/amdgpu/CMakeLists.txt
@@ -22,7 +22,7 @@ endif()
 # as of rocm-3.7, hsa is installed with cmake packages and kmt is found via hsa
 find_package(hsa-runtime64 QUIET 1.2.0 HINTS ${CMAKE_INSTALL_PREFIX} PATHS /opt/rocm)
 
-if(NOT CMAKE_SYSTEM_PROCESSOR MATCHES "(x86_64)|(ppc64le)|(aarch64)$" AND CMAKE_SYSTEM_NAME MATCHES "Linux")
+if(NOT (CMAKE_SYSTEM_PROCESSOR MATCHES "(x86_64)|(ppc64le)|(aarch64)$" AND CMAKE_SYSTEM_NAME MATCHES "Linux"))
   libomptarget_say("Not building AMDGPU plugin: only support AMDGPU in Linux x86_64, ppc64le, or aarch64 hosts")
   return()
 endif()
